internal/store/horizontalpodautoscaler.go

```diff
@@ -180,10 +188,10 @@ func createHPASpecMinReplicas() generator.FamilyGenerator {
 	)
 }
 
-func createHPASpecTargetMetric() generator.FamilyGenerator {
+func createHPASpecTargetContainerMetric() generator.FamilyGenerator {
 	return *generator.NewFamilyGeneratorWithStability(
-		"kube_horizontalpodautoscaler_spec_target_metric",
```

Comment:
Yeah, this is going to be problematic. Here's the thing with stability guarantees in KSM:
* ALPHA: labelsets may be dropped or added to the metric,
* BETA: labelsets may only be added to the metric,
* STABLE: labelsets are frozen.

However, removing a metric itself, needs to undergo a deprecation cycle. In order to facilitate migration to patterns that avoid the recent "duplication" errors [post-Promtheus v2.52.0](https://github.com/prometheus/prometheus/issues/14089) we probably should to gate all such metrics behind a, say, `--legacy-metrics` and state that these may be duplicated, disabled by default.

@mrueg WDYT?